### ISSUE TITLE
"Use next to skip iteration" reporting wrong location.

### ISSUE THREAD

**Issue by zenspider** [bbatsov/rubocop#2138]
Related to #1238:

```ruby
def xxx
  loop do
    do_stuff

    unless condition then
      do_other_stuff1
      do_other_stuff2
      do_other_stuff3
    end
  end
end
```

reports:

```
Offenses:

wtf.rb:2:3: C: Use next to skip iteration.
  loop do
  ^^^^
```

But, the `loop` on line 2 column 3 is not the problem. The problem is in the contents of the loop, on line 5, column 4. This makes the report confusing and misleading.

**Comment by bbatsov** [comment]
Any suggestions about the position of the offence and its message?

**Comment by zenspider** [comment]
I would expect the `^^^` part to be pointing at the jumping construct (`if`/`unless`), not the looping construct.

**Comment by rrosenblum** [comment]
I agree with @zenspider on this one. The error should be pointing to the location of the offense (`unless condition then`). If we move the the offense to point to the condition, then we can have the cop call out multiple offense in a given loop.

Example
```ruby
def xxx 
  loop do
    do_stuff

    unless condition then
      do_other_stuff1
      do_other_stuff2
      do_other_stuff3
    end 

    unless condition2 then
      do_other_stuff4
      do_other_stuff5
      do_other_stuff6
    end 
  end 
end
```
reports
```
test.rb:2:3: C: Use next to skip iteration.
  loop do
  ^^^^
```
With the current implementation, this could lead to confusion for someone trying to fix what they think is only one issue, but in actuality is two issues.

**Comment by rrosenblum** [comment]
I just started working on this cop. I realized that the example that I wrote earlier only contains 1 offense, not 2 like I originally was thinking. I am still going to work on moving the highlight.
